src/components/TodoContainer.js

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from "react";
+import TodoList from "./TodoList";
+import AddTodoForm from "./AddTodoForm";
+import styles from "../TodoListItem.module.css";
+import PropTypes from "prop-types";
+
+export default function TodoContainer() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchData = async () => {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+    };
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME} `;
+
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+      const data = await response.json();
+
+      const todos = data.records.map((data) => {
+        const newTodo = {
+          title: data.fields.title,
+          id: data.id,
+        };
+        return newTodo;
+      });
+
+      setTodoList(todos);
+      setIsLoading(false);
+    } catch (error) {
+      console.log(error.message);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const addTodo = async (newTodo) => {
+    try {
+      const airtableData = {
+        fields: {
+          title: newTodo.title,
+        },
+      };
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        },
+        body: JSON.stringify(airtableData),
+      };
+      const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME} `;
+
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+      const jsonResponse = await response.json();
+
+      const airtableTodo = {
+        title: jsonResponse.fields.title,
+        id: jsonResponse.id,
+      };
+
+      console.log(airtableTodo);
+      setTodoList([...todoList, airtableTodo]);
+    } catch (error) {
+      console.log(error.message);
+    }
+  };
+  const removeTodo = (id) => {
+    const removeItem = todoList.filter((todo) => todo.id !== id);
+
+    setTodoList(removeItem);
+  };
+
+  return (
+    <div className={styles.Container}>
+      <h1 className={styles.MainHeader}>Todo List</h1>
+
+      <AddTodoForm onAddTodo={addTodo} />
+      {isLoading ? (
+        <p className={styles.Loading}>Loading...</p>
+      ) : (
+        <TodoList todoList={todoList} removeTodo={removeTodo} />
+      )}
+    </div>
+  );
```

Comment:
for using this `Container` pattern, you've done the right thing by moving all of this out of `App.jsx`. But there's one more step for this pattern as I'm familiar with it: 

Make sure the `TodoContainer` only returns a single thing, which is the new component `TodoView`. The `Container` can handle all the fetching and related states to that, and then the `View` will be completely responsible for rendering any visuals. 
```suggestion
  return  <TodoView
       onAddTodo={addTodo}
       onRemoveTodo={removeTodo]
       isLoading={isLoading}
    />;
```

Then the `TodoView` component could have no state in it, and these 3 props above, and render all the visuals that we removed from the container 

```javascript
  return (
    <div className={styles.Container}>
      <h1 className={styles.MainHeader}>Todo List</h1>
      <AddTodoForm onAddTodo={addTodo} />
      {isLoading ? (
        <p className={styles.Loading}>Loading...</p>
      ) : (
        <TodoList todoList={todoList} removeTodo={removeTodo} />
      )}
    </div>
  );
```

Reply:
This would also mean that neither `TodoContainer` nor `App` have any need to import the `styles`